src/js/providers/tracks-mixin.js

```diff
@@ -459,11 +459,13 @@ function addTextTracks(tracksArray) {
                 (vttCues) => {
                     this.addVTTCuesToTrack(textTrackAny, vttCues);
                 },
-                (error) => {
-                    this.trigger(ERROR, {
-                        message: 'Captions failed to load',
-                        reason: error
-                    });
+                (key, file, url, error) => {
+                    if (error.code === 404) {
```

Comment:
if `error.code` is 404, we should set the reason to `'File not found'`, but if the error code is anything else we still want to trigger the error event